### PR TITLE
inital commit for yellowhammer jupyter assistant

### PR DESCRIPTION
**Pull request by yue-here**
Initial implementation of %%llm line magic with datalab API prompt for datalab-LLM interactions

### CODE REVIEW

File: src/yellowhammer/test.ipynb
Comment (ml-evs) [inline src/yellowhammer/test.ipynb:1]: Could we add these cases as a simple pytest test that runs in the CI? It should already all be set up and ready to go

File: src/yellowhammer/datalab-api-prompt.md
Comment (ml-evs) [inline src/yellowhammer/datalab-api-prompt.md:1]: I think we should move this out to the top-level of the repo in a `./prompts` folder, along with the system prompt

File: examples/Demo.ipynb
Comment (ml-evs) [inline examples/Demo.ipynb:1]: Tbh I would just replace the notebook I made with this!

File: src/yellowhammer/llm.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from langchain_core.prompts import ChatPromptTemplate
+from pydantic import BaseModel, Field
+from .prompt import SYSTEM_PROMPT
+from typing import Union
+
+# Pydantic schema to use with .with_structured_output()
+class code(BaseModel):
```
Comment (ml-evs) [inline src/yellowhammer/llm.py:7]: the linter will probably complain about classes that are not in CamelCase

File: src/yellowhammer/magics.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        api_key=env.get("LLM_API_KEY"),
+        api_model=env.get("LLM_MODEL", None),
+        api_temperature=env.get("LLM_TEMP", temp),
+    )
+
+    return agent_executor.invoke({"context": API_PROMPT, "messages": messages})
+
+# Class to manage state and expose the main magics
+@magics_class
+class datalabMagics(Magics):
```
Comment (ml-evs) [inline src/yellowhammer/magics.py:36]: same here with camel case

**Review by ml-evs — CHANGES_REQUESTED** [review]
Hi @yue-here, great work! I haven't got time to try it out right now but hopefully can give it a go tomorrow. I have a few surface-level comments that you can either address, or we can just fix-up after merging, otherwise nice stuff!

I think the main thing is just: run the linter yourself via pre-commit (`pre-commit run --all-files` should fix most things), and fix any uncorrectable suggestions, then also just convert your test notebook into a pytest test that can actually run in the CI.

Once we get to it, we can also have the CI directly execute notebooks to make sure the magics work.

File: src/yellowhammer/datalab-api-prompt.md
Comment (yue-here) [reply src/yellowhammer/datalab-api-prompt.md:1]: moved

File: examples/Demo.ipynb
Comment (yue-here) [reply examples/Demo.ipynb:1]: Will do, is "001-Title" the naming convention?

File: src/yellowhammer/magics.py (hunk 2 above)
Comment (yue-here) [reply src/yellowhammer/magics.py:36]: Will convert all to camelcase (with the initial cap)

**Review by ml-evs — APPROVED** [review]
Thanks for this @yue-here! This works well for me locally, so let's merge and figure out some details later. In particular, I want to try an example that interacts with block data and writes some non-trivial analysis code.

**Pull request merged**